client/js/_author/components/assessments/question_types/drag_and_drop/drop_zone.jsx

```diff
@@ -0,0 +1,245 @@
+          topPos: newTop,
+          leftPos: newLeft,
+        });
+        break;
+      case 'topRight':
+        this.setState({
+          topPos: newTop,
+          rightPos: newRight,
+        });
+        break;
+      case 'bottomLeft':
+        this.setState({
+          bottomPos: newBottom,
+          leftPos: newLeft,
+        });
+        break;
+      case 'bottomRight':
+        this.setState({
+          bottomPos: newBottom,
+          rightPos: newRight,
+        });
+        break;
+      default:
+        break;
+    }
+  }
+
+  moveZone(x, y) {
+    const { leftPos, topPos, rightPos, bottomPos, initialX, initialY } = this.state;
+    const target = this.props.target.getBoundingClientRect();
+    if (!x || !y) { return; }
+
+    const deltaX = x - initialX;
+    const deltaY = y - initialY;
+
+    // TODO: this will warp the zone if you hit an edge, should fix that
+    this.setState({
+      leftPos: DropZone.boundaryCheck(leftPos + deltaX, target.right - target.left),
+      topPos: DropZone.boundaryCheck(topPos + deltaY, target.bottom - target.top),
+      rightPos: DropZone.boundaryCheck(rightPos + deltaX, target.right - target.left),
+      bottomPos: DropZone.boundaryCheck(bottomPos + deltaY, target.bottom - target.top),
+      initialX: x,
+      initialY: y,
+    });
+  }
+
+  updateZone() {
+    const { leftPos, topPos, rightPos, bottomPos } = this.state;
+
+    this.props.editZone(this.props.zone.id, {
+      xPos: leftPos,
+      yPos: topPos,
+      height: bottomPos - topPos,
+      width: rightPos - leftPos,
+    });
+  }
+
+  // TODO: Some of this could be extracted to css, other parts not so much
+  styles() {
+    const manipulators = {
+      display: this.props.isActive ? '' : 'none',
+    };
+
+    return {
+      positioning: {
+        position: 'absolute',
+        height: '10px',
+        width: '10px',
+        border: '1px solid black',
+        backgroundColor: 'white',
+        ...manipulators,
+      },
+      topLeftSelector: {
+        top: '-5px',
+        left: '-5px',
+        cursor: 'nwse-resize',
+      },
+      topRightSelector: {
+        top: '-5px',
+        right: '-5px',
+        cursor: 'nesw-resize',
+      },
+      bottomLeftSelector: {
+        bottom: '-5px',
+        left: '-5px',
+        cursor: 'nesw-resize',
+      },
+      bottomRightSelector: {
+        bottom: '-5px',
+        right: '-5px',
+        cursor: 'nwse-resize',
+      },
+      middleSelector: {
+        position: 'absolute',
+        cursor: 'move',
+        top: '5px',
+        left: '5px',
+        width: 'calc(100% - 10px)',
+        height: 'calc(100% - 10px)',
+        ...manipulators,
+      },
+      manipulators,
+    };
+  }
+
+  render() {
+    const { zone } = this.props;
+    const styles = this.styles();
+
+    return (
+      <div
+        className="au-c-drop-zone au-c-zone1 is-active"
+        style={this.zonePosition(zone)}
+        onClick={this.props.setActive}
+      >
+        <div
```

Comment:
These could probably be generated by iterating over an array of topLeft, bottomLeft, etc